### PR TITLE
create performanceProfile during init

### PR DESCRIPTION
**Pull request by xrajesh**
Epic: https://issues.redhat.com/browse/ACM-4082
Story: https://issues.redhat.com/browse/ACM-4493

### Description of changes
- Create PerformanceProfile in Kruize before starting the server

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,4 +7,7 @@ require (
 	k8s.io/klog/v2 v2.90.1
 )
 
-require github.com/go-logr/logr v1.2.3 // indirect
+require (
+	github.com/go-logr/logr v1.2.3 // indirect
+	k8s.io/klog v1.0.0 // indirect
```
Comment (SherinV) [inline go.mod:12]: We use k8s.io/klog/v2 in other search repos. Maybe good to keep it consistent for CVEs and other stuff.

File: main.go [hunk 2]
```diff
@@ -24,5 +26,21 @@ func main() {
...
 	}
 
+	// Load PerformanceProfile in Kruize Instance first
+	perfProfileInitialized := false
+	for !perfProfileInitialized {
+		klog.Info("Initializing performanceProfile.")
+		perfProfileInitialized = kruize.InitPerformanceProfile()
+		if perfProfileInitialized {
+			klog.Info("Initialized performanceProfile.")
+			break
```
Comment (SherinV) [inline main.go:36]: We need to check periodically if this is still true., in case the kruize container dies in between.

**Review by SherinV — APPROVED** [review]

File: main.go (hunk 2 above)
Comment (xrajesh) [reply main.go:36]: @SherinV Dev Preview : We will check during getPerformanceProfile 
                   Tech Preview : We need a initContainer for Kruize , which will ensure the presence of PerformanceProfile

**Pull request merged**